<h2>Model Development</h2>

In [1]:
import pandas as pd
import numpy as np
import matplotlib as plt
path = 'https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/automobileEDA.csv'
df = pd.read_csv(path)
df.head()

symboling  normalized-losses         make aspiration num-of-doors  \
0          3                122  alfa-romero        std          two   
1          3                122  alfa-romero        std          two   
2          1                122  alfa-romero        std          two   
3          2                164         audi        std         four   
4          2                164         audi        std         four   

    body-style drive-wheels engine-location  wheel-base    length  ...  \
0  convertible          rwd           front        88.6  0.811148  ...   
1  convertible          rwd           front        88.6  0.811148  ...   
2    hatchback          rwd           front        94.5  0.822681  ...   
3        sedan          fwd           front        99.8  0.848630  ...   
4        sedan          4wd           front        99.4  0.848630  ...   

   compression-ratio  horsepower  peak-rpm city-mpg highway-mpg    price  \
0                9.0       111.0    5000.0       21          27  13495.0   
1                9.0       111.0    5000.0       21          27  16500.0   
2                9.0       154.0    5000.0       19          26  16500.0   
3               10.0       102.0    5500.0       24          30  13950.0   
4                8.0       115.0    5500.0       18          22  17450.0   

  city-L/100km  horsepower-binned  diesel  gas  
0    11.190476             Medium       0    1  
1    11.190476             Medium       0    1  
2    12.368421             Medium       0    1  
3     9.791667             Medium       0    1  
4    13.055556             Medium       0    1  

[5 rows x 29 columns]

<h2>Linear Regression and Multiple Linear Regression</h2>

In [2]:
from sklearn.linear_model import LinearRegression
lm = LinearRegression()
lm
X = df[['highway-mpg']]
Y = df[['price']]
lm.fit(X, Y)
lm.intercept_

array([38423.30585816])

In [3]:
lm.coef_

array([[-821.73337832]])

In [4]:
y = lm.intercept_ + lm.coef_*25
y

array([[17879.97140011]])

<h3>What is the final estimated linear model we get?</h3>

plugging in the actual values we get
<h4> price = 38423.31 - 821.73 * highway-mpg</h4>

In [5]:
# Create a linear regression object
lm1 = LinearRegression()

In [6]:
# Train the model using 'engine-size' as the independent variable on 'price' as the dependent variable
lm1.fit(df[["engine-size"]], df[["price"]])

LinearRegression()

In [7]:
# Find the slope and intercept of the model
lm1.intercept_

array([-7963.33890628])

In [8]:
lm1.coef_

array([[166.86001569]])

In [9]:
# What is the equation of the predicted line
# price = -7963.34 + 166.86 * engine-size

<h2>Multiple Linear Regression</h2>

What if we want to predict car price using more than one variable ?


If we want to use more variables in our model to predict car price, we can use <b>Multiple linear Regression</b>.Multiple Linear Regression is very simple to Simple Linear Regression, but this method is used to explain the realtionship between one continuos response reponse (dependent) variable and <b>two or more</b> predictor (independent) variables. Most of the real-world regression models involve multiple predictors. We will illustrate the structure by using the structure by using four predictor variables, but these results can generalize to any integer.

In [10]:
from sklearn.linear_model import LinearRegression
lm3 = LinearRegression()
lm3
X = df[['horsepower', 'curb-weight', 'engine-size', 'highway-mpg']]
Y = df[['price']]
lm3.fit(X, Y)
lm3.intercept_

array([-15806.62462633])

In [11]:
lm3.coef_

array([[53.49574423,  4.70770099, 81.53026382, 36.05748882]])

<h2>The Predicted model is: </h2>

<b>price = -15806.62 + 53.50 * horsepower + 4.71 * curb-weight + 81.53 * endgine-size + 36.06 * highway-mpg</b>

In [13]:
# Create and train a multiple linear regression model
# where the response variable is price and the predictor variable is "normalized-losses" and "highway-mpg"

<h2>Model Evaluation using Visualization</h2>